### PR TITLE
Fix cannot parse LEFT JOIN correctly

### LINKED ISSUE #40: Cannot parse the LEFT JOIN expression
It seems that the join syntax is not supported, here is a simple test

```sql
with t1 as ( select 1 as value ), t2 as ( select 2 as value ) select * from t1 left join t2 on true
```

```
line 0:84 <EOF> or ';' was expected, but got: "join"
with t1 as ( select 1 as value ), t2 as ( select 2 as value ) select * from t1 left join t2 on true
```

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #40

### CODE REVIEW

File: parser/parser_common.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ func (p *Parser) wrapError(err error) error {
 			for j := 0; j < column; j++ {
 				buf.WriteByte(' ')
 			}
-			buf.WriteString(strings.Repeat("^", len(p.lexer.lastToken.String)))
+			//buf.WriteString(strings.Repeat("^", len(p.lexer.lastToken.String)))
```
Comment (Lance726) [inline parser/parser_common.go:317]: Is it expected here?

Comment (git-hulk) [reply parser/parser_common.go:317]: No, it's a typo.

Comment (git-hulk) [inline parser/parser_common.go:317]:
```suggestion
			buf.WriteString(strings.Repeat("^", len(p.lexer.lastToken.String)))
```

Comment (Lance726) [inline parser/parser_common.go]:
```suggestion
			buf.WriteString(strings.Repeat("^", len(p.lexer.lastToken.String)))
```

**Review by Lance726 — APPROVED** [review]

**Pull request merged**